sys/include/net/ng_udp.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_net_udp UDP
+ * @ingroup     net
+ * @brief       RIOT's implementation of the UDP protocol
+ *
+ * @{
+ *
+ * @file
+ * @brief       UDP interface definition
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NG_UDP_H_
+#define NG_UDP_H_
+
+#include <stdint.h>
+
+#include "byteorder.h"
+#include "net/ng_netbase.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Default message queue size for the UDP thread
+ */
+#ifndef NG_UDP_MSG_QUEUE_SIZE
+#define NG_UDP_MSG_QUEUE_SIZE   (8U)
+#endif
+
+/**
+ * @brief   Priority of the pktdump thread
+ */
+#ifndef NG_UDP_PRIO
+#define NG_UDP_PRIO             (PRIORITY_MAIN - 2)
+#endif
+
+/**
+ * @brief   Default stack size to use for the UDP thread
+ */
+#ifndef NG_UDP_STACK_SIZE
+#define NG_UDP_STACK_SIZE       (KERNEL_CONF_STACKSIZE_DEFAULT)
+#endif
+
+/**
+ * @brief   UDP header definition
+ */
+typedef struct __attribute__((packed)) {
+    network_uint16_t src_port;      /**< source port, in network byte order */
+    network_uint16_t dst_port;      /**< destination port, network byte order */
+    network_uint16_t length;        /**< payload length (including the header),
+                                     *   network byte order */
+    network_uint16_t checksum;      /**< checksum */
+} ng_udp_hdr_t;
+
+/**
+ * @brief   Calculate the checksum for the given packet
+ *
+ * @param[in] hdr           Pointer to the UDP header
+ * @param[in] pseudo_hdr    Pointer to the network layer header
+ *
+ * @return  0 on success
+ * @return  -EBADMSG if @p pkt is not of type @ref NG_NETTYPE_UDP
```

Comment:
Doxygen has a problem finding the reference to `NG_NETTYPE_UDP`, since `MODULE_NG_UDP` in its scope is not defined.